### PR TITLE
GitHub Release Provenance

### PR DESCRIPTION
**Pull request by CodeGat**
In this PR:
* Added workflows to validate the versions of spack-packages and spack-config used
* Added config files for versioning

Should close access-nri/build-cd#24 !

### CODE REVIEW

File: config/versions.schema.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+            "type": "string"
+        },
+        "spack-packages": {
+            "type": "string"
+        },
+        "spack-config": {
+            "type": "string"
+        }
+    }
+}
```
Comment (jo-basevi) [inline config/versions.schema.json:17]: Should this be more strict?
```
"required": [ "$schema", "spack-packages", "spack-config" ],
 "additionalProperties": false
```

**Review by jo-basevi — COMMENTED** [review]
Just a couple of questions/edits then happy to re-review

File: config/versions.schema.json (hunk 1 above)
Comment (CodeGat) [reply config/versions.schema.json:19]: Done, thank you :)

**Review by jo-basevi — APPROVED** [review]
LGTM

**Pull request merged**